ingestion/src/metadata/ingestion/source/database/salesforce/metadata.py

```diff
@@ -79,6 +79,8 @@ def __init__(self, config, metadata: OpenMetadata):
         self.client = get_connection(self.service_connection)
         self.table_constraints = None
         self.database_source_state = set()
+        self.table_desc_map = {}
```

Comment:
I think we don't need to store the data as part of the object. From what I am seeing it is  used in `yield_table`. You can probably just return the data from `get_table_description` move the logic of `set_table_description` in `get_table_description` and get rid of `set_table_description`.

Reply:
@TeddyCr 
We get table descriptions for all tables in salesforce at once, not for each table. If we moved the logic of `set_table_description` into `get_table_description`, I think that it would be a redundant process as `self.client.toolingexecute(...)` would be executed as many times as the number of tables to get the descriptions of all the tables each time. Is there a better method to avoid executing `self.client.toolingexecute(...)` many times?

Reply:
@TeddyCr 
As discussed in Slack, we have modified it to get the description for each table.
The reason for the fix is as follows.
- We want to split the results to consider the possibility that there is a limit on the number of results to be retrieved by the API.
- It seems that using the LIMIT and OFFSET clauses would not return correct results due to [SOQL Operation Limitations](https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/reference_objects_soql_limits.htm).